docs/proposals/20210716-async-azure-resource-creation-deletion.md

```diff
@@ -0,0 +1,382 @@
+    // PrivateDNSFailedCondition means the private DNS failed to be created.
+    PrivateDNSFailedCondition clusterv1.ConditionType = "PrivateDNSFailed"
+
+    // BastionHostReadyCondition means the bastion host exists and is ready to be used.
+    BastionHostReadyCondition clusterv1.ConditionType = "BastionHostReady"
+    // BastionHostCreatingReason means the bastion host is being created.
+    BastionHostCreatingReason clusterv1.ConditionType = "BastionHostCreating"
+    // BastionHostDeletingReason means the bastion host is being deleted.
+    BastionHostDeletingReason clusterv1.ConditionType = "BastionHostDeleting"
+    // BastionHostFailedCondition means the bastion host failed to be created.
+    BastionHostFailedCondition clusterv1.ConditionType = "BastionHostFailed"
+
+    // InboundNATRulesReadyCondition means the inbound NAT rules exist and are ready to be used.
+    InboundNATRulesReadyCondition clusterv1.ConditionType = "InboundNATRulesReady"
+    // InboundNATRulesCreatingReason means the inbound NAT rules are being created.
+    InboundNATRulesCreatingReason clusterv1.ConditionType = "InboundNATRulesCreating"
+    // InboundNATRulesDeletingReason means the inbound NAT rules are being deleted.
+    InboundNATRulesDeletingReason clusterv1.ConditionType = "InboundNATRulesDeleting"
+    // InboundNATRulesFailedCondition means the inbound NAT rules failed to be created.
+    InboundNATRulesFailedCondition clusterv1.ConditionType = "InboundNATRulesFailed"
+
+    // AvailabilitySetReadyCondition means the availability set exists and is ready to be used.
+    AvailabilitySetReadyCondition clusterv1.ConditionType = "AvailabilitySetReady"
+    // AvailabilitySetCreatingReason means the availability set is being created.
+    AvailabilitySetCreatingReason clusterv1.ConditionType = "AvailabilitySetCreating"
+    // AvailabilitySetDeletingReason means the availability set is being deleted.
+    AvailabilitySetDeletingReason clusterv1.ConditionType = "AvailabilitySetDeleting"
+    // AvailabilitySetFailedCondition means the availability set failed to be created.
+    AvailabilitySetFailedCondition clusterv1.ConditionType = "AvailabilitySetFailed"
+
+    // RoleAssignmentReadyCondition means the role assignment exists and is ready to be used.
+    RoleAssignmentReadyCondition clusterv1.ConditionType = "RoleAssignmentReady"
+    // RoleAssignmentCreatingReason means the role assignment is being created.
+    RoleAssignmentCreatingReason clusterv1.ConditionType = "RoleAssignmentCreating"
+    // RoleAssignmentDeletingReason means the role assignment is being deleted.
+    RoleAssignmentDeletingReason clusterv1.ConditionType = "RoleAssignmentDeleting"
+    // RoleAssignmentFailedCondition means the role assignment failed to be created.
+    RoleAssignmentFailedCondition clusterv1.ConditionType = "RoleAssignmentFailed"
+)
+```
+
+### <a name='OpenQuestions'></a>Open Questions
```

Comment:
Another question: how do we handle reconciler error so that transient "operation not done" errors, which are fully expected in this new model, do not pollute manager logs?

Reply:
added